library/std/src/sys/xous/thread.rs

```diff
@@ -68,14 +68,20 @@ impl Thread {
         )
         .map_err(|code| io::Error::from_raw_os_error(code as i32))?;
 
-        extern "C" fn thread_start(main: *mut usize, guard_page_pre: usize, stack_size: usize) {
-            unsafe {
-                // Finally, let's run some code.
-                Box::from_raw(main as *mut Box<dyn FnOnce()>)();
+        extern "C" fn thread_start(
+            main: *mut usize,
+            guard_page_pre: usize,
+            stack_size: usize,
+        ) -> ! {
+            {
+                unsafe {
+                    // Run the contents of the new thread.
+                    Box::from_raw(main as *mut Box<dyn FnOnce()>)();
+                }
             }
```

Comment:
Why this extra block around the unsafe block?

Reply:
Because `Box::from_raw()` is unsafe, and without the `unsafe {}` block it doesn't compile:

```
   Compiling std v0.0.0 (/opt/Xous/rust-next-next/library/std)
warning: dropping unsupported crate type `dylib` for target `riscv32imac-unknown-xous-elf`

error[E0133]: call to unsafe function is unsafe and requires unsafe block
  --> library/std/src/sys/xous/thread.rs:77:13
   |
22 |     pub unsafe fn new(stack: usize, p: Box<dyn FnOnce()>) -> io::Result<Thread> {
   |     --------------------------------------------------------------------------- items do not inherit unsafety from separate enclosing items
...
77 |             Box::from_raw(main as *mut Box<dyn FnOnce()>)();
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior

For more information about this error, try `rustc --explain E0133`.
warning: `std` (lib) generated 1 warning
error: could not compile `std` (lib) due to 1 previous error; 1 warning emitted
Build completed unsuccessfully in 0:00:06
```

Reply:
I mean why `{ unsafe { ... } }` instead of `unsafe { ... }`?

Reply:
I'm sorry, I completely missed that. Yes, that's a very good catch. Let me remove the extra braces.